### DISCUSSION TITLE
请问终端上传提示"sha" wasn't supplied.什么情况？手动上传是OK的

### DISCUSSION THREAD

**Discussion by luxuxl** [gee1k/uPic#238] [Q&A]
如图😣<img width="1112" alt="image" src="https://user-images.githubusercontent.com/111971103/234537020-c5664111-c7d1-42ef-ac4f-f7b4c80dadca.png">

**Comment by baotiao** [comment]
应该是repo 已经有一个重名的文件了, 去github 上面删除一下就可以了.

**Comment by tisfeng** [comment]
我也在用 uPic， 由于作者一直没有回复，我就自己 fork 优化修复这个问题 https://github.com/gee1k/uPic/pull/251 ，并使用新的签名发布了一个更新版本 [1.4.3](https://github.com/tisfeng/uPic/releases/tag/1.4.3) ，有需要的请自取。

更新内容如下：

---- 

## 1.4.3 💎 优化上传图片名字重复问题 sha wasn't supplied

之前版本如果上传图片的名字重复，会报错提示 sha wasn't supplied https://github.com/gee1k/uPic/discussions/238 ，此次更新做了如下优化：

- 如果上传的图片名已存在，且 sha 值相同，则直接返回该图片的 URL。
- 如果上传的图片名已存在，但 sha 值不同（即不同的图片，只是同名），则以随机图片名再次上传。
